### PR TITLE
Add description of Pythnet to docs

### PR DESCRIPTION
**Pull request by jayantk**
also some links to that description

### CODE REVIEW

File: consumers/on-demand.md [hunk 1]
```diff
@@ -27,6 +27,9 @@ The on-demand model has several benefits over the push model:
 - **Low latency** -- Every transaction can use a recent off-chain price, instead of relying on the last on-chain update pushed by the oracle itself.
 - **More price feeds** -- Pyth Network can scale to thousands of price feeds due to its gas efficiency.
   The oracle incurs no added costs for each additional feed, and users pay gas costs for new feeds only when those feeds are used on-chain.
+- **Reliable in volatile conditions** -- Push oracles can fail to land price updates in volatile market conditions.
+  In these conditions, the oracle competes with other more valuable transactions (such as DEX trades or liquidations) for bandwidth, and often cannot pay enough to land price updates.
+  This problem does not occur in the on-demand model, as Pyth price updates are incorporated into the valuable transactions themselves.
```
Comment (jayantk) [inline consumers/on-demand.md:32]: drive-by comment

Comment (ali-bahjati) [reply consumers/on-demand.md:32]: It's a very nice one! 👍
